signcrypt_seal.go

```diff
@@ -69,6 +69,7 @@ func (sss *signcryptSealStream) signcryptBytes(b []byte) error {
 
 	nonce := nonceForChunkSigncryption(sss.numBlocks)
 
+	// TODO: Encode payload chunk.
```

Comment:
Yeah, I think I'm coming around to your last suggestion of have an "associated data" bundle, rather than making the final flag part of the plaintext (i.e. the input to Seal)

The disadvantage with making the final flag part of the plaintext (and not repeating it) is that it is impossible to determine the end of a saltpack message if you're not able to decrypt it. If we don't care about that, then we can make it part of the plaintext (for encrypt mode, too) and everything will work as-is. Otherwise, then we'd have to treat it as associated data.